File: hw/tipantiza/u3/hw22MazeSolidPrinciples/MazeSystem_py/model/room.py
class Room:

    def __init__(self, x, y):
        self.x = x
        self.y = y
        self._is_entrance = False
        self._is_exit = False
        self.connecting_doors = []

    def get_connecting_doors(self):
        return self.connecting_doors

    def add_door(self, door):
        self.connecting_doors.append(door)

    def get_x(self):
        return self.x

    def get_y(self):
        return self.y

    def is_entrance(self):
        return self._is_entrance

    def set_entrance(self, entrance):
        self._is_entrance = entrance

    def is_exit(self):
        return self._is_exit

    def set_exit(self, exit_room):
        self._is_exit = exit_room

    def __str__(self):
        return (
            f"Room({self.x}, {self.y}) "
            f"[Entrance: {self._is_entrance}, Exit: {self._is_exit}]"
        )

File: hw/tipantiza/u3/hw22MazeSolidPrinciples/MazeSystem_java/MazeSystem/src/main/java/ec/edu/espe/mazesystem/model/Room.java
package ec.edu.espe.mazesystem.model;  
import java.util.ArrayList;
import java.util.List;

public class Room {
    private int x;
    private int y;
    private boolean isEntrance;
    private boolean isExit;
    private List<Door> connectingDoors;

    public Room(int x, int y) {
        this.x = x;
        this.y = y;
        this.isEntrance = false;
        this.isExit = false;
        this.connectingDoors = new ArrayList<>();
    }

    public List<Door> getConnectingDoors() {
        return connectingDoors;
    }

    public void addDoor(Door door) {
        connectingDoors.add(door);
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    public boolean isEntrance() {
        return isEntrance;
    }

    public void setEntrance(boolean entrance) {
        isEntrance = entrance;
    }

    public boolean isExit() {
        return isExit;
    }

    public void setExit(boolean exit) {
        isExit = exit;
    }

    @Override
    public String toString() {
        return String.format("Room(%d, %d) [Entrance: %s, Exit: %s]", 
            x, y, isEntrance, isExit);
    }
}